app/src/main/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt

```diff
@@ -1,22 +1,44 @@
 package org.fossasia.openevent.general.search
 
+import android.Manifest
+import android.annotation.SuppressLint
 import android.content.Intent
+import android.content.pm.PackageManager
+import android.location.Address
+import android.location.Geocoder
 import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
+import android.support.v4.app.ActivityCompat
+import android.support.v4.content.ContextCompat
 import org.fossasia.openevent.general.R
 
 import android.support.v7.widget.SearchView
 import android.view.MenuItem
+import android.view.View
+import android.widget.Toast
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.location.LocationServices
 import kotlinx.android.synthetic.main.activity_search_location.*
+import kotlinx.android.synthetic.main.fragment_login.*
 import org.fossasia.openevent.general.MainActivity
 import org.koin.android.architecture.ext.viewModel
+import java.io.IOException
+import java.util.*
 
 private const val FROM_SEARCH: String = "FromSearchFragment"
 private const val TO_SEARCH: String = "ToSearchFragment"
 
 class SearchLocationActivity : AppCompatActivity() {
 
     private val searchLocationViewModel by viewModel<SearchLocationViewModel>()
+    var latitude: Double = 0.0
+    var longitude: Double = 0.0
+    val MY_PERMISSIONS_REQUEST = 1000
+    var city = "Singapore"
+    val locationRequest: LocationRequest = LocationRequest.create()
+    var locationCallback = LocationCallback()
```

Comment:
Why creating a new one when it is replaced in onCreate?

Reply:
LocationRequest and Location callback are also being used in onRequestPermissionResult.

Reply:
But why initializing it here?

Reply:
So should I remove initialization?